inc/wp-rest-api-extensions.php

```diff
@@ -0,0 +1,218 @@
+			$image_src = wp_get_attachment_image_src( $image_id, $size );
+			if ( ! $image_src ) {
+				continue;
+			}
+			$size_data['source_url'] = $image_src[0];
+		}
+	} elseif ( is_string( $featured_image['media_details'] ) ) {
+		// This was added to work around conflicts with plugins that cause
+		// wp_get_attachment_metadata() to return a string.
+		$featured_image['media_details'] = new stdClass();
+		$featured_image['media_details']->sizes = new stdClass();
+	} else {
+		$featured_image['media_details']['sizes'] = new stdClass();
+	}
+
+	return apply_filters( 'uds_rest_api_featured_image', $featured_image, $image_id );
+}
+
+/**
+ * Register the ACF post author fields to add them to the /post API endpoint.
+ *
+ * @see https://since1979.dev/add-custom-acf-fields-to-the-wp-rest-api/
+ * @uses register_rest_field() https://developer.wordpress.org/reference/functions/register_rest_field/
+ * @uses array() https://www.php.net/manual/en/function.array.php
+ */
+function uds_register_news_author_api_fields() {
+	register_rest_field(
+		'post',
+		'uds_news_author',
+		array(
+			'get_callback' => 'uds_get_news_author_api_field',
+			'schema' => null,
+		)
+	);
+}
+add_action( 'rest_api_init', 'uds_register_news_author_api_fields' );
+
+/**
+ * Fetch and return the value of the uds_news_author ACF fields.
+ *
+ * @param   object $post      The Post object.
+ *
+ * @see https://since1979.dev/add-custom-acf-fields-to-the-wp-rest-api/
+ * @uses get_field() https://www.advancedcustomfields.com/resources/get_field/
+ */
+function uds_get_news_author_api_field( $post ) {
+	$uds_news_author_name = get_field( 'uds_news_author_name', $post['id'] );
+	$uds_news_author_title = get_field( 'uds_news_author_title', $post['id'] );
+	$uds_news_author_email = get_field( 'uds_news_author_email', $post['id'] );
+	$uds_news_author_phone = get_field( 'uds_news_author_phone', $post['id'] );
+
+	if (
+		empty( $uds_news_author_name ) &&
+		empty( $uds_news_author_title ) &&
+		empty( $uds_news_author_email ) &&
+		empty( $uds_news_author_phone )
+	) {
+		return null;
+	}
+
+	return array(
+		'name' => $uds_news_author_name,
+		'title' => $uds_news_author_title,
+		'email' => $uds_news_author_email,
+		'phone' => $uds_news_author_phone,
+	);
+}
+
+/**
+ * Register the ACF Story Hero fields for the /post API endpoint.
+ *
+ * @see https://since1979.dev/add-custom-acf-fields-to-the-wp-rest-api/
+ * @uses register_rest_field() https://developer.wordpress.org/reference/functions/register_rest_field/
+ * @uses array() https://www.php.net/manual/en/function.array.php
+ */
+function uds_register_post_story_hero_api_fields() {
+	register_rest_field(
+		'post',
+		'uds_story_hero',
+		array(
+			'get_callback' => 'uds_get_post_story_hero_api_field',
+			'schema' => null,
+		)
+	);
+}
+add_action( 'rest_api_init', 'uds_register_post_story_hero_api_fields' );
+
+/**
+ * Fetch and return the value of the uds_news_author ACF fields.
```

Comment:
Oops! I forgot to update this function comment!

Reply:
Fixed. :-)